src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -214,8 +208,31 @@ public override bool SendChunked
         /// <returns>Returns the value of the header with the given name.</returns>
         public override string GetHeader(string headerName)
         {
-            Util.CheckArgumentNullAndEmpty(headerName, "headerName");
-            return HttpWebRequestMessage.GetHeaderValue(this.httpRequest, headerName);
+            if (_contentHeaderNames.Contains(headerName))
+            {
+                // If this is a "Content" header then we retrieve the value either
+                // from the message content (if available) or the cache (otherwise)
+                if (_requestMessage.Content != null)
+                {
+                    return string.Join(",", _requestMessage.Content.Headers.GetValues(headerName));
+                }
+
+                string headerValue;
+                if (string.Equals(headerName, XmlConstants.HttpContentType, StringComparison.OrdinalIgnoreCase))
+                {
+                    return _contentHeaderValueCache.TryGetValue("Content-Type", out headerValue) ? headerValue : string.Empty;
+                }
+                else if (string.Equals(headerName, XmlConstants.HttpContentLength, StringComparison.OrdinalIgnoreCase))
+                {
+                    return _contentHeaderValueCache.TryGetValue("Content-Length", out headerValue) ? headerValue : string.Empty;
+                }
+                else
+                {
+                    return _contentHeaderValueCache.TryGetValue(headerName, out headerValue) ? headerValue : string.Empty;
+                }               
+            }
+

```

Comment:
The Contains in the line below, confirm that it's not case-sensitive

Reply:
it is not.